### PR TITLE
fix: Invalid URL endpoint propagation

### PR DESCRIPTION
**Pull request by TeodorSAP**
## Description

Changes proposed in this pull request (what was done and why):

- Implement endpoint validator
- UTs for the validator
- validate trace-, metric-, and log-pipelines
- E2E tests for the validation scenarios

Changes refer to particular issues, PRs or documents:

- Resolves https://github.com/kyma-project/telemetry-manager/issues/1270

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] If the change is user-facing, the documentation has been adjusted.
- [x] The feature is unit-tested.
- [x] The feature is e2e-tested.

### CODE REVIEW

**Review by NHingerl — DISMISSED** [review]

File: internal/conditions/conditions.go [hunk 1]
```diff
@@ -57,6 +58,7 @@ const (
 )
 
 var commonMessages = map[string]string{
+	ReasonEndpointInvalid:         "Endpoint invalid: %s",
```
Comment (skhalash) [inline internal/conditions/conditions.go:61]: Endpoint invalid sounds too generic. Maybe let's have separate messages for LogPipeline (HTTP output host invalid) and MetricPipeline & Trace Pipeline (OTLP output endpoint invalid)

File: internal/reconciler/logpipeline/validator.go [hunk 2]
```diff
@@ -33,6 +38,12 @@ func (v *Validator) validate(ctx context.Context, pipeline *telemetryv1alpha1.Lo
 		return err
 	}
 
+	if pipeline.Spec.Output.Custom == "" {
```
Comment (skhalash) [inline internal/reconciler/logpipeline/validator.go:41]: Why not directly check for HTTP output?

File: internal/reconciler/metricpipeline/validator.go [hunk 3]
```diff
@@ -27,6 +32,10 @@ func (v *Validator) validate(ctx context.Context, pipeline *telemetryv1alpha1.Me
 		return err
 	}
 
+	if err := v.EndpointValidator.Validate(ctx, &pipeline.Spec.Output.Otlp.Endpoint); err != nil {
+		return err
+	}
+
```
Comment (skhalash) [inline internal/reconciler/metricpipeline/validator.go:38]: Let's add a check for nillable fields

File: internal/reconciler/tracepipeline/validator.go [hunk 4]
```diff
@@ -27,6 +32,10 @@ func (v *Validator) validate(ctx context.Context, pipeline *telemetryv1alpha1.Tr
 		return err
 	}
 
+	if err := v.EndpointValidator.Validate(ctx, &pipeline.Spec.Output.Otlp.Endpoint); err != nil {
+		return err
+	}
+
```
Comment (skhalash) [inline internal/reconciler/tracepipeline/validator.go:38]: Let's add a check for nillable fields

**Review by skhalash — CHANGES_REQUESTED** [review]

File: internal/reconciler/logpipeline/validator.go (hunk 2 above)
Comment (TeodorSAP) [reply internal/reconciler/logpipeline/validator.go:41]: See: `test/e2e/logs_custom_output_test.go`

Comment (TeodorSAP) [reply internal/reconciler/logpipeline/validator.go]: I misunderstood what `custom` does, and omitted the `loki` output. Implementing your review comment...

**Review by NHingerl — APPROVED** [review]

**Review by skhalash — APPROVED** [review]

**Pull request merged**